### PR TITLE
Add basic SuSE support

### PR DESCRIPTION
**Pull request by tuxmea**

### CODE REVIEW

**Review by rwaffen — COMMENTED** [review]
LGTM

File: manifests/service.pp [hunk 1]
```diff
@@ -7,6 +7,11 @@
 ) {
   assert_private()
 
+  if $facts['os']['family'] == 'Suse' {
+    exec { "${gitlab_ci_runner::binary_path} install -u ${gitlab_ci_runner::user}":
```
[the remark this answers is not in the document]
Comment (tuxmea) [reply manifests/service.pp:11]: No. My test installation did not require a separate `systemctl daemon-reload`

File: manifests/service.pp [hunk 2]
```diff
@@ -7,6 +7,11 @@
 ) {
   assert_private()
 
+  if $facts['os']['family'] == 'Suse' {
```
[the remark this answers is not in the document]
Comment (tuxmea) [reply manifests/service.pp:10]: Both. Or even better: this method works for any OS! I am about to implement the any OS usage in a next PR

File: manifests/init.pp [hunk 3]
```diff
@@ -89,6 +101,12 @@
   Optional[String]                           $sentry_dsn        = undef,
   Optional[Pattern[/.*:.+/]]                 $listen_address    = undef,
   Optional[Gitlab_ci_runner::Session_server] $session_server    = undef,
+  Enum['repo', 'binary']                     $install_method    = 'repo',
+  Stdlib::HTTPUrl                            $binary_source     = 'https://s3.dualstack.us-east-1.amazonaws.com/gitlab-runner-downloads/latest/binaries/gitlab-runner-linux-amd64',
```
[the remark this answers is not in the document]
Comment (tuxmea) [reply manifests/init.pp:105]: What about people with local http server without https? The idea is to let people override the url.

**Review by rwaffen — APPROVED** [review]

File: manifests/init.pp (hunk 3 above)
[the remark this answers is not in the document]
Comment (bastelfreak) [reply manifests/init.pp:105]: Do we still need to support plaintext HTTP in 2024? :(

File: README.md [hunk 4]
```diff
@@ -69,6 +72,17 @@ gitlab_ci_runner::runners:
     ensure: absent
 ```
 
+## SLES
+
+There are no gitlab_ci_runner repositories for SLES/zypper available!
+Please set the following data to be able to use this module on SLES:
+
+```yaml
+gitlab_ci_runner::install_method: 'binary' # required for SLES
```
Comment (bastelfreak) [inline README.md:81]: please document that the `puppet/archive` module is required for this method.

File: manifests/init.pp (hunk 3 above)
[the remark this answers is not in the document]
Comment (tuxmea) [reply manifests/init.pp:105]: Yes. Customer insists on using an internal HTTP only server. :-(

File: README.md (hunk 4 above)
Comment (tuxmea) [reply README.md:84]: Done

**Review by bastelfreak — APPROVED** [review]

**Pull request merged**